src/main/java/com/libertymutualgroup/herman/cli/Cli.java

```diff
@@ -0,0 +1,91 @@
+
+import com.amazonaws.regions.Regions;
+import com.libertymutualgroup.herman.logging.SysoutLogger;
+import com.libertymutualgroup.herman.task.cli.ecs.ECSPushTask;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+@Command(description = "Runs Herman the AWS Task Helper", name = "herman", mixinStandardHelpOptions = true)
+public class Cli implements Callable<Void> {
+
+    @CommandLine.Parameters(paramLabel = "TASK", description = "Task to be executed", arity = "1")
+    private Tasks task;
+
+    @Option(names = {"-d", "--directory"}, description = "Path to configuration files", showDefaultValue = CommandLine.Help.Visibility.ALWAYS)
+    private String rootPath = System.getProperty("user.dir");
+
+    @Option(names = {"-e", "-env", "--environment"}, description = "Environment to deploy")
+    private String environmentName;
+
+    @Option(names = {"-t", "--timeout"}, description = "Task timeout (in minutes)", showDefaultValue = CommandLine.Help.Visibility.ALWAYS)
+    private int timeout = 5;
+
+    @Option(names = {"-r", "--region"}, description = "AWS Region to perform tasks", showDefaultValue = CommandLine.Help.Visibility.ALWAYS, arity = "1")
+    private Regions region = Regions.US_EAST_1;
+
+    @Option(names = {"-v", "-vars", "--variables"}, description = "Custom build variables to be injected. <KEY>==<VALUE>")
+    private Map<String, String> customVariables = new HashMap<>();
+
+    public static void main(String[] args) {
+        CommandLine.call(new Cli(), args);
+    }
+
+    @Override public Void call() throws Exception {
+        SysoutLogger logger = new SysoutLogger();
+        if (task.equals(Tasks.CFT_PUSH)) {
+            logger.addErrorLogEntry("CFT Push not yet implemented in CLI");
+        }
+        String absPath = new File(this.rootPath).getAbsolutePath();
+
+        switch (task) {
+            case ECS_PUSH:
+                logger.addLogEntry("Starting ECS Push");
+                ECSPushTask ecsPush = new ECSPushTask(logger);
+                ecsPush.doExecute(absPath, timeout, environmentName, region, customVariables);
+                break;
+            case CFT_PUSH:
+                logger.addErrorLogEntry("Not yet implemented in CLI");
+                System.exit(1);
+                break;
+            case ECR_REPO_CREATE:
+                logger.addErrorLogEntry("Not yet implemented in CLI");
+                System.exit(1);
+                break;
+            case LAMBDA_CREATE:
+                logger.addErrorLogEntry("Not yet implemented in CLI");
+                System.exit(1);
+                break;
+            case NEW_RELIC_DEPLOYMENT:
+                logger.addErrorLogEntry("Not yet implemented in CLI");
+                System.exit(1);
+                break;
+            case S3_CREATE:
```

Comment:
It shouldn't ever occur, but do you want to add a default case?

Reply:
Since task is an enum, Picocli will throw an error if you pass an invalid task before this even executes. That being said, it's probably best practice to add one, so I'll go ahead and do it anyway just in case.